### DISCUSSION TITLE
Test failure vs locale

### DISCUSSION THREAD

**Discussion by christf** [resteasy/resteasy#4022] [General]
On a machine with german localization I have just attempted to run the tests.
They are failing in https://github.com/resteasy/resteasy/blob/main/resteasy-core-spi/src/test/java/org/jboss/resteasy/spi/config/SizeUnitTestCase.java#L39 because with german localization settings,
1.5KB actually is formatted as 1,5 KB.

In an attempt to work around this, I found adding -Duser.country=US -Duser.language=en to  MAVEN_OPTS does not help, but sure enough, changing the desired outcome to "1,5KB" in the above test case, avoids the problem.

Is it desired tests should only work on machines with C/en locale?

**Comment by jamezp** [comment]
That's a good catch. I think we should allow this to work with any locale. I filed https://issues.redhat.com/browse/RESTEASY-3449. I think we should one allow a `Locale` to be passed to the `SizeUnit.toString(long)` method and the `SizeUnit.toHumanReadable(long)` method. I also think we need to allow the tests to pass on different locales.

Thank you for finding this!
